packages/core/src/scope.ts

```diff
@@ -469,7 +472,7 @@ export class Scope implements ScopeInterface {
   }
 
   /** @inheritDoc */
-  public getScopeData(): ScopeData {
+  public getPerScopeData(): ScopeData {
```

Comment:
Does `getPerScopeData` have to be public API?

I also am not the biggest fan of the scope class being aware of a separate global scope. I'd rather it be the consumers of the scope that decide to use the global scope.

Reply:
Hmm, I see your point. I'll see if we can do this another way!